### PR TITLE
Add new PA-specific Common Alignment Producers

### PR DESCRIPTION
**Pull request by jrcastle**
Adds in new PA-specific common alignment producers for muon-based tracker alignment collections. In pA and AA collisions, the yields of isoMus and Z's are strongly suppressed by the *RelCombIsoMuons step. Studies on 2013 pPb data show improvement in alignment yields using producers where this step has been removed as outlined in the following presentation at the tracker alignment weekly meeting:

https://indico.cern.ch/event/536888/contributions/2326743/attachments/1348506/2034547/10_5_16_Alignment.pdf

Additional edits were made to the new Millepede configuration files to include these new track collections in the standard workflows.

These new ALCA producers were then added to wf 281.

### CODE REVIEW

**Comment by mmusich** [comment]
please test

File: Alignment/CommonAlignmentProducer/python/ALCARECOTkAlMuonIsolatedHI_cff.py [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
     )
 
 ALCARECOTkAlMuonIsolatedHIDCSFilter = ALCARECOTkAlMuonIsolatedDCSFilter.clone()
 
-ALCARECOTkAlMuonIsolatedHIGoodMuons = ALCARECOTkAlMuonIsolatedGoodMuons.clone()
-ALCARECOTkAlMuonIsolatedHIRelCombIsoMuons = ALCARECOTkAlMuonIsolatedRelCombIsoMuons.clone(
-    src = 'ALCARECOTkAlMuonIsolatedHIGoodMuons'
-)
-
 ALCARECOTkAlMuonIsolatedHI = ALCARECOTkAlMuonIsolated.clone(
```
Comment (mmusich) [inline Alignment/CommonAlignmentProducer/python/ALCARECOTkAlMuonIsolatedHI_cff.py]: @jrcastle with this being remove you cannot let it in the path: https://github.com/cms-sw/cmssw/pull/16155/files#diff-9a1d4158ab5e79a6d2b1fe963ef4712cR29

resulting in:

```
raceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/2016-42/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-10-09-0000/bin/slc6_amd64_gcc530/cmsDriver.py", line 55, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/2016-42/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-10-09-0000/bin/slc6_amd64_gcc530/cmsDriver.py", line 27, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-10-09-2300/python/Configuration/Applications/ConfigBuilder.py", line 2052, in prepare
    self.addStandardSequences()
  File "/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-10-09-2300/python/Configuration/Applications/ConfigBuilder.py", line 745, in addStandardSequences
    getattr(self,"prepare_"+stepName)(sequence = '+'.join(stepSpec))
  File "/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-10-09-2300/python/Configuration/Applications/ConfigBuilder.py", line 1240, in prepare_ALCA
    alcaConfig=self.loadDefaultOrSpecifiedCFF(sequence,self.ALCADefaultCFF)
  File "/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-10-09-2300/python/Configuration/Applications/ConfigBuilder.py", line 1192, in loadDefaultOrSpecifiedCFF
    l=self.loadAndRemember(defaultCFF,unsch)
  File "/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-10-09-2300/python/Configuration/Applications/ConfigBuilder.py", line 325, in loadAndRemember
    self.process.load(includeFile)
  File "/cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-10-09-2300/python/FWCore/ParameterSet/Config.py", line 519, in load
    module = __import__(moduleName)
  File "/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-09-2300/python/Configuration/StandardSequences/AlCaRecoStreams_cff.py", line 140, in <module>
    from DQMOffline.Configuration.AlCaRecoDQM_cff import *
  File "/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-09-2300/python/DQMOffline/Configuration/AlCaRecoDQM_cff.py", line 8, in <module>
    from DQMOffline.Alignment.ALCARECOTkAlDQM_cff import *
  File "/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-09-2300/python/DQMOffline/Alignment/ALCARECOTkAlDQM_cff.py", line 487, in <module>
    from Alignment.CommonAlignmentProducer.ALCARECOTkAlMuonIsolatedHI_cff import ALCARECOTkAlMuonIsolatedHIHLT
  File "/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-09-2300/python/Alignment/CommonAlignmentProducer/ALCARECOTkAlMuonIsolatedHI_cff.py", line 30, in <module>
    +ALCARECOTkAlMuonIsolatedHI
NameError: name 'ALCARECOTkAlMuonIsolatedHIGoodMuons' is not defined

```

Comment (jrcastle) [inline Alignment/CommonAlignmentProducer/python/ALCARECOTkAlMuonIsolatedHI_cff.py]: Ah, that was a stupid mistake that I should have caught before committing.  I made a fix and I'm testing locally before I update the branch.

**Comment by mmusich** [comment]
@jrcastle ready for tests?

**Comment by jrcastle** [comment]
Yes, please test.

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
@jrcastle since #16122 is merged can you backport this one including the change of the matrix, when the next IB appears?

**Comment by jrcastle** [comment]
Sure thing, I'll keep my eye out for the next IB.  I'm assuming it will be the 23:00 release.

**Comment by mmusich** [comment]
so running 281.0 (default n. events=10) after having merged this PR gives me:
```
$ edmEventSize -v -a TkAlMuonIsolatedPA.root 
File TkAlMuonIsolatedPA.root Events 1
Branch Name | Average Uncompressed Size (Bytes/Event) | Average Compressed Size (Bytes/Event) 
SiPixelClusteredmNewDetSetVector_ALCARECOTkAlMuonIsolatedPA__RECO. 28104 28104
recoTrackExtras_ALCARECOTkAlMuonIsolatedPA__RECO. 26870 26820
recoTracks_ALCARECOTkAlMuonIsolatedPA__RECO. 25691 25590
SiStripClusteredmNewDetSetVector_ALCARECOTkAlMuonIsolatedPA__RECO. 20337 20326
recoVertexsedmAssociation_offlinePrimaryVertices__RECO. 20228 19934
recoVertexs_offlinePrimaryVertices__RECO. 14400 12550
DcsStatuss_scalersRawToDigi__RECO. 10717 10717
L1AcceptBunchCrossings_scalersRawToDigi__RECO. 6237 6237
TrackingRecHitsOwned_ALCARECOTkAlMuonIsolatedPA__RECO. 5405 5008
EventProductProvenance 4237 2711
intedmValueMap_offlinePrimaryVertices__RECO. 2760 2500
floatedmValueMap_offlinePrimaryVertices__RECO. 2492 2492
L1GlobalTriggerReadoutRecord_gtDigis__RECO. 2496 2466
edmTriggerResults_TriggerResults__RECO. 2423 2420
edmTriggerResults_TriggerResults__HLT. 2407 2404
edmTriggerResults_TriggerResults__SIM. 2395 2394
EventAuxiliary 738 204
EventSelections 683 165
BranchListIndexes 640 114
```
```
$ edmEventSize -v -a TkAlMinBias.root 
File TkAlMinBias.root Events 10
Branch Name | Average Uncompressed Size (Bytes/Event) | Average Compressed Size (Bytes/Event) 
recoTrackExtras_ALCARECOTkAlMinBias__RECO. 7378.3 5719.1
recoTracks_ALCARECOTkAlMinBias__RECO. 7280.2 4518.7
SiStripClusteredmNewDetSetVector_ALCARECOTkAlMinBias__RECO. 7104.3 3965.2
SiPixelClusteredmNewDetSetVector_ALCARECOTkAlMinBias__RECO. 4940.3 3719
TrackingRecHitsOwned_ALCARECOTkAlMinBias__RECO. 18748.4 2860.5
recoBeamSpot_offlineBeamSpot__RECO. 2603.9 2360.5
recoVertexsedmAssociation_offlinePrimaryVertices__RECO. 2284.5 2029.9
EventProductProvenance 2099.6 1538.3
recoVertexs_offlinePrimaryVertices__RECO. 3404.3 1533
DcsStatuss_scalersRawToDigi__RECO. 1115.8 1091.9
L1AcceptBunchCrossings_scalersRawToDigi__RECO. 646.2 634.3
intedmValueMap_offlinePrimaryVertices__RECO. 535.9 280.2
floatedmValueMap_offlinePrimaryVertices__RECO. 299.5 260.3
L1GlobalTriggerReadoutRecord_gtDigis__RECO. 315.3 250.9
edmTriggerResults_TriggerResults__RECO. 325.1 246.8
edmTriggerResults_TriggerResults__HLT. 321.7 245
edmTriggerResults_TriggerResults__SIM. 309.7 243.9
EventAuxiliary 193.5 31.8
EventSelections 134 21.4
BranchListIndexes 82 15.2
```

while the Z->mm and the Upsilon producers yield no event.
I would say it is in the expected ballpark

**Comment by dmitrijus** [comment]
+1

**Comment by fabozzi** [comment]
+1

**Comment by mmusich** [comment]
+1 
(formally OK with the changes)

**Pull request merged**